dispatcher/dispatcher.go

```diff
@@ -0,0 +1,140 @@
+package dispatcher
+
+import (
+	"sync"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+)
+
+// Event is a generic structure for adding events to the Event Store
+type Event interface {
```

Comment:
We need some notion of `EntityID`, and event `Type` at the very least to satisfy the CQRS/Flux pattern. I'm not sure those properties need to belong on the core `Event` type, so perhaps we just have a dispatcher `Event` that wraps the core `Event`? Thoughts?